### ISSUE TITLE
cargo install should use --cap-lints

### ISSUE THREAD

**Issue by whitequark** [rust-lang/cargo#3453]
I was installing `cargo edit` today, and this is how it failed:
```
error: unnecessary qualification
  --> /home/whitequark/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-edit-0.1.3/src/manifest.rs:13:1
   |
13 | quick_error! {
   | ^
   |
note: lint level defined here
  --> /home/whitequark/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-edit-0.1.3/src/lib.rs:4:176
   |
4  | #![deny(missing_docs, missing_debug_implementations, missing_copy_implementations, trivial_casts, trivial_numeric_casts, unsafe_code, unstable_features, unused_import_braces, unused_qualifications)]
   |                                                                                                                                                                                ^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in a macro outside of the current crate
```
This seems undesirable.

**Comment by alexcrichton** [comment]
Makes sense to me!

**Comment by sanmai-NL** [comment]
To clarify: info on [`rustc --cap-lints`](https://github.com/rust-lang/rfcs/blob/master/text/1193-cap-lints.md).

**Comment by dwijnand** [comment]
Fixed as of ~1.15 (perhaps in #3221).
